src/network_manager.rs

```diff
@@ -29,6 +29,21 @@ pub struct ActiveConnections {
     pub connections: HashMap<PeerId, PeerConnection>,
     pub listeners: HashMap<SocketAddr, TransportType>,
 }
+
+impl ActiveConnections {
+    /// Check if an address has to be refused or not
+    pub fn check_addr_accepted(&self, addr: &SocketAddr) -> bool {
+        if self.connections.is_empty() {
+            true
+        } else {
+            !self
+                .connections
+                .iter()
+                .any(|(_, connection)| connection.endpoint.get_target_addr().ip() != addr.ip())
```

Comment:
`any()` return true if one element match the predicate. Here for example if the first address on `ActiveConnections` is `127.0.0.1` and the connecting address is `127.0.0.2`. Your any will return `true` and so your function `false` but the function should return `true`.

I think you wanted to use `all` not equal or `any` equal.

Reply:
What I meant in this line is:
"If the new  peer has an IP address that matches ANY of the existing connections's IP address, refuse it`
In that sense, I iterate over all connections, and if the IP address of the connecting peer matches one in the connections, the function returns `false`, as "it is not accepted"

Reply:
If the connecting IP is `127.0.0.2` and there is an existing connection to `127.0.0.1`, then `any` will return `false`, and the function will return `true`, accepting this new connection

Reply:
No your `any()` will return true because in this case `connection.endpoint.get_target_addr().ip() != addr.ip()` is true that's what I wanted to point out